File: Easy/3340_Check_Balanced_String/Solution.py



class Solution:
    def isBalanced(self, num: str) -> bool:
        even_sum = 0
        odd_sum = 0
        
        for i, digit in enumerate(num):
            if i % 2 == 0:
                even_sum += int(digit)
            else:
                odd_sum += int(digit)
        
        return even_sum == odd_sum


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.isBalanced("1234"))  # Output: False

    # test cases 2
    print(solution.isBalanced("24123"))  # Output: True

File: Easy/3340_Check_Balanced_String/Solution.go
package main

import "fmt"

func isBalanced(num string) bool {
	evenSum, oddSum := 0, 0
	for i, ch := range num {
		digit := int(ch - '0')
		if i%2 == 0 {
			evenSum += digit
		} else {
			oddSum += digit
		}
	}
	return evenSum == oddSum
}

func main() {
	// Test case 1
	fmt.Println(isBalanced("1234")) // Expected: false

	// Test case 2
	fmt.Println(isBalanced("24123")) // Expected: true

}
